corehq/apps/custom_data_fields/models.py

```diff
@@ -74,8 +74,8 @@ def validate_regex(self, value):
             return _("'{value}' is not a valid match for {label}").format(
                 value=value, label=self.label)
 
-    def validate_required(self, value):
-        if self.is_required and not value:
+    def validate_required(self, value, field_view):
```

Comment:
Not blocking at all since I know this isn't part of your changes but I don't think I fully understand the logic of passing the field value (?) into this argument

Reply:
`value` is needed to know whether or not something is defined for this field during bulk upload.

It should return the error message only if the field is required and there is no defined value (aka `value` is an empty string). If the field is required and `value` is a non empty string, then `not value` would be False, so there would be no error (which is the correct behavior).